hack/generators/controllers/networking/main.go

```diff
@@ -89,7 +89,7 @@ var inputControllersNeeded = &typesNeeded{
 		URL:                               "networking.k8s.io",
 		CacheType:                         "IngressV1beta1",
 		AcceptsIngressClassNameAnnotation: true,
-		AcceptsIngressClassNameSpec:       false,
+		AcceptsIngressClassNameSpec:       true,
```

Comment:
Is this and the related changes elsewhere actually what we want (ditto for the `extensions` version)? Those iterations of Ingress did not support the `IngressClassName` field and needed the annotation.

If the existing tests are using `IngressClassName` we'd need to create parallel tests that use the older API version and set the annotation rather than adding support for `IngressClassName` to the old API version controllers.

Reply:
As it turns out some late revisions of these APIs _did_ support the spec field (thus why it's available in the Go API).

Enabling this is only done for completeness: the use case where you would be stuck on a legacy version of the API but would also have the latest version of it including this field is extremely limited, but conversely it doesn't actually _hurt_ anything to add this either as older versions of the API will simply drop the unknown field on post.